pyfar/classes/coordinates.py

```diff
@@ -98,50 +95,266 @@ def __init__(self, points_1=None, points_2=None, points_3=None,
             ``""``, which initializes an empty string.
         """
 
-        if points_1 is None:
-            points_1 = []
-        if points_2 is None:
-            points_2 = []
-        if points_3 is None:
-            points_3 = []
-
-        # init emtpy object
+        # init empty object
         super(Coordinates, self).__init__()
 
-        # set the coordinate system
-        self._system = self._make_system(domain, convention, unit)
+        # test Deprecation warning
+        if domain != 'cart' or convention is not None or unit is not None:
+            warnings.warn((
+                "This function will be changed in pyfar 0.8.0 to "
+                "init(x, y, z)."),
+                    PyfarDeprecationWarning)
 
-        # save coordinates to self
-        self._set_points(points_1, points_2, points_3, True)
+        # set the coordinate system
+        system = self._make_system(domain, convention, unit)
+        self._system = system
+
+        # set coordinates according to system
+        if domain == 'cart':
+            self._set_points(points_1, points_2, points_3)
+        elif domain == 'sph':
+            self._set_sph(
+                points_1, points_2, points_3,
+                system['convention'], system['unit'])
+        elif domain == 'cyl':
+            self._set_cyl(
+                points_1, points_2, points_3,
+                system['convention'], system['unit'])
+        else:
+            raise ValueError(
+                f"Domain for {domain} is not implemented.")
 
         # save meta data
         self._set_weights(weights)
         self._sh_order = sh_order
         self._comment = comment
 
-    def set_cart(self, points_1, points_2, points_3,
-                 convention='right', unit='met'):
+        if sh_order is not None:
+            warnings.warn((
+                "This function will be deprecated in pyfar 0.8.0 in favor "
+                "of SamplingSphere."),
+                    PyfarDeprecationWarning)
+
+    @classmethod
+    def from_cartesian(
+            cls, x, y, z, weights: np.array = None, comment: str = ""):
+        """Create a Coordinates class object from a set of points in the
+        right-handed cartesian coordinate system. See
+        :py:mod:`coordinates concepts <pyfar._concepts.coordinates>` for
+        more information.
+
+        Parameters
+        ----------
+        x : ndarray, double
+            x-coordinate
+        y : ndarray, double
+            y-coordinate
+        z : ndarray, double
+            z-coordinate
+        weights: array like, number, optional
+            weighting factors for coordinate points. The `shape` of the array
+            must match the `shape` of the individual coordinate arrays.
+            The default is ``None``.
+        comment : str, optional
+            comment about the stored coordinate points. The default is
+            ``""``, which initializes an empty string.
+        """
```

Comment:
Since this is a class method, it might not be clear to less experiencied users how to call it. Should we have a small example for all `from_*` methods, e.g., `coordinates = pyfar.Coordinates().from_cartesian(1, 0, 0)`?